src/porepy/numerics/ad/equation_system.py

```diff
@@ -1236,6 +1236,47 @@ def remove_equation(self, name: str) -> Operator | None:
         else:
             raise ValueError(f"Cannot remove unknown equation {name}")
 
+    def update_equation(
+        self,
+        equation_name: str,
+        new_equation: Operator,
+        grids: Optional[DomainList] = None,
+        equations_per_grid_entity: Optional[dict[GridEntity, int]] = None,
+    ) -> None:
+        """Updates an existing equation with a new equation operator.
+
+        This method removes the existing equation and sets a new equation under the same
+        name as the old equation.
+
+        Parameters:
+            equation_name: Name of the equation to be updated.
+            new_equation: New equation in AD form.
+            grids: A list of subdomain *or* interface grids on which the equation is
+                defined. The default value is None, and in that case, the grids of the
+                previous equation are used.
+            equations_per_grid_entity: a dictionary describing how many equations
+                ``equation_operator`` provides. This is a temporary work-around until
+                operators are able to provide information on their image space. The
+                dictionary must contain the number of equations per grid entity (cells,
+                faces, nodes) for the operator. The default value is None, and in that
+                case, the equations_per_grid_entity of the previous equation are used.
+
+        """
+        if grids is None:
+            grids = list(  # type: ignore
```

Comment:
Can the ignore be replaced by a declaration? That might allow us to skip the ignore below.